README.md

```diff
@@ -51,28 +51,87 @@ Here is a simple example of how to use web authentication with DRAW:
+  final userAgent = 'foobar';
+  final configUri = Uri.parse('draw.ini');
+
+  // Create a `Reddit` instance using a configuration file in the
+  // current directory.
+  final reddit = await Reddit.createInstance(userAgent: userAgent,
+                                             configUri: configUri);
 
-  // Build the URL used for authentication. See `WebAuthenticator` documentation for parameters.
+  // Build the URL used for authentication. See `WebAuthenticator`
+  // documentation for parameters.
   final auth_url = reddit.auth.url(['*'], 'foobar'));
   
   // ...
-  // Complete authentication at `auth_url` in the browser and retrieve the `code` query
-  // parameter from the redirect URL.
+  // Complete authentication at `auth_url` in the browser and retrieve
+  // the `code` query parameter from the redirect URL.
   // ...
 
-  // Assuming the `code` query parameter is stored in a variable `auth_code`, we pass it to the
-  // `authorize` method in the `WebAuthenticator`.
+  // Assuming the `code` query parameter is stored in a variable
+  // `auth_code`, we pass it to the `authorize` method in the
+  // `WebAuthenticator`.
   await reddit.auth.authorize(auth_code);
 
-  // If everything worked correctly, we should be able to retrieve information about the authenticated
-  // account.
+  // If everything worked correctly, we should be able to retrieve
+  // information about the authenticated account.
+  print(await reddit.user.me());
+}
+```
+
+It is also possible to restore cached credentials in order to avoid the need to complete the web authentication flow on each run:
+
+```dart
+import 'package:draw/draw.dart';
+
+// Provides methods to load and save credentials.
+import 'credential_loader.dart';
+
+main() async {
+  final userAgent = 'foobar';
+  final configUri = Uri.parse('draw.ini');
+
+  // Load cached credentials from disk, if available.
+  final credentialsJson = await loadCredentials();
+
+  var reddit;
+
+  if (credentialsJson == null) {
+    reddit =
+        await Reddit.createInstance(userAgent: userAgent,
+                                    configUri: configUri);
+
+    // Build the URL used for authentication. See `WebAuthenticator`
+    // documentation for parameters.
+    final auth_url = reddit.auth.url(['*'], 'foobar');
+
+    // ...
+    // Complete authentication at `auth_url` in the browser and retrieve
+    // the `code` query parameter from the redirect URL.
+    // ...
+
+    // Assuming the `code` query parameter is stored in a variable
+    // `auth_code`, we pass it to the `authorize` method in the
+    // `WebAuthenticator`.
+    await reddit.auth.authorize(auth_code);
+
+    // Write credentials to disk.
+    await writeCredentials(reddit.auth.credentials.toJson());
+  } else {
+    // Create a new Reddit instance using previously cached credentials.
+    reddit = await Reddit.restoreAuthenticatedInstance(
```

Comment:
So I read up on **Futures and Error Handling**, but I can't seem to find how we would handle a future that throws, when we have await.

So my question is:
1. Should we have a `try catch` here or does the _Future_ return `null`, when it resolves with an error during the use of `await`?

https://www.dartlang.org/guides/libraries/futures-error-handling

Reply:
The `Future` isn't completed when `await` is used and an exception is thrown (or if it is, it doesn't matter anymore) which can be caught in a `catch`. I don't think we should need one here for just the examples.